cli/npm/managed/resolvers/common/lifecycle_scripts.rs

```diff
@@ -155,6 +157,28 @@ impl<'a> LifecycleScripts<'a> {
     packages: &[NpmResolutionPackage],
     root_node_modules_dir_path: &Path,
     progress_bar: &ProgressBar,
+  ) -> Result<(), AnyError> {
+    let kill_signal = KillSignal::default();
+    run_future_with_kill_signal(
+      kill_signal.clone(),
+      self.finish_with_cancellation(
+        snapshot,
+        packages,
+        root_node_modules_dir_path,
+        progress_bar,
+        kill_signal,
+      ),
+    )
+    .await
```

Comment:
What's done in this PR is the signal listening is setup at a high level and a `KillSignal` from `deno_task_shell` is piped through all the code into the task runner. Then when a signal is received it's sent along the `KillSignal` (which is kind of like a `CancellationToken`, but for sending signals).